numpy/lib/_datasource.py

```diff
@@ -38,13 +38,61 @@
 import os
 import sys
 import shutil
+import io
 
 _open = open
 
+def _python2_bz2open(fn, mode, encoding, newline):
+    """ wrapper to open bz2 in text mode """
+    import bz2
+
+    if "t" in mode:
+        raise ValueError("bz2 text files not supported in python2")
```

Comment:
This should probably have a:

```
if "b" in mode:
    raise ValueError("Invalid mode: %r" % (mode,))
```

In fact, can this validation be extracted to a helper function, since it appears twice?